hibernate-core/src/main/java/org/hibernate/boot/archive/internal/JRTFileBasedArchiveDescriptor.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later.
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.boot.archive.internal;
+
+import org.hibernate.boot.archive.spi.*;
+
+import java.io.*;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.util.Enumeration;
+import java.util.jar.JarFile;
+import java.util.jar.JarInputStream;
+import java.util.zip.ZipEntry;
+
+import static org.hibernate.internal.log.UrlMessageBundle.*;
+
+/**
+ * An ArchiveDescriptor implementation leveraging the {@link java.util.jar.JarFile} API for processing - specifically meant to support the new URL format for JRT and modules
+ *
+ * @author Steve Ebersole
+ * @author Marc Magon
+ */
+public class JRTFileBasedArchiveDescriptor extends AbstractArchiveDescriptor {
+	/**
+	 * Constructs a JarFileBasedArchiveDescriptor
+	 *
+	 * @param archiveDescriptorFactory The factory creating this
+	 * @param archiveUrl The url to the JRT Module VFS file
+	 * @param entry The prefix for entries within the JRT VFS url
+	 */
+	public JRTFileBasedArchiveDescriptor(
+			ArchiveDescriptorFactory archiveDescriptorFactory,
+			URL archiveUrl,
+			String entry) {
+		super( archiveDescriptorFactory, archiveUrl, entry );
+	}
+
+	@Override
+	public void visitArchive(ArchiveContext context) {
+		final String jrtPart = getArchiveUrl().getPath();
```

Comment:
Ok the main problem here is that you've been using syntax and JDK APIs which won't compile with Java 8: this branch of Hibernate ORM still aims to be compatible with Java 8 users, which implies we run the build with Java 8, only integration tests are run with a matrix of multiple JDKs.

We have some options:

- use a multi-release jar (not my favourite, tools don't really deal well with those)
- we merge this in ORM 6 exclusively, assuming @sebersole and the other developers agree to drop Java 8 compatibility (I was going to open this dicussion anyway as there are more benefits into this, but it implies it won't make it in 5.x version and you'll have to wait a bit longer to have this)
- perhaps it's possible for you to use JDK8 compatible APIs? I only see errors caused by `FileVisitResult`, `SimpleFileVisitor`, `Paths` ... inconvenient but don't seem stricly related to needing Jigsaw related APIs.

Reply:
Understood